engine/src/test/java/org/operaton/bpm/engine/impl/jobexecutor/setInitialRetries/LegacyJobDeclarationRetriesTest.java

```diff
@@ -0,0 +1,145 @@
+      .configurator(config -> config.setLegacyJobRetryBehaviorEnabled(true))
+      .build();
+
+  @RegisterExtension
+  ProcessEngineTestRule testRule = new ProcessEngineTestRule(engineRule);
+
+  private ManagementService managementService;
+  private RuntimeService runtimeService;
+
+  @BeforeEach
+  void init() {
+    this.managementService = engineRule.getManagementService();
+    this.runtimeService = engineRule.getRuntimeService();
+  }
+
+  @Test
+  void testRetryTimeCycleWithZeroRetriesAndFailure() {
+    // given
+    String retryInterval = "R0/PT5M";
+    String processDefinitionName = "testRetryTimeCycleWithZeroRetriesAndFailure";
+    BpmnModelInstance bpmnModelInstance = getBpmnModelInstance(processDefinitionName, retryInterval);
+    testRule.deploy(bpmnModelInstance);
+
+    // when
+    String processInstanceId = runtimeService.startProcessInstanceByKey(processDefinitionName).getId();
+    Job job = managementService.createJobQuery().processInstanceId(processInstanceId).singleResult();
+
+    // then
+    assertThat(job.getRetries()).isEqualTo(3);
+
+    // when
+    try {
+      managementService.executeJob(job.getId());
+    } catch (Exception e) {
+      // ignore
+    }
+    job = managementService.createJobQuery().processInstanceId(processInstanceId).singleResult();
+
+    // then
+    assertThat(job.getRetries()).isZero();
+  }
+
+  @Test
+  void testRetryTimeCycleWithFailure() {
+    // given
+    String retryInterval = "R5/PT5M";
+    String processDefinitionName = "testRetryTimeCycleWithFailure";
+    BpmnModelInstance bpmnModelInstance = getBpmnModelInstance(processDefinitionName, retryInterval);
+    testRule.deploy(bpmnModelInstance);
+
+    // when
+    String processInstanceId = runtimeService.startProcessInstanceByKey(processDefinitionName).getId();
+    Job job = managementService.createJobQuery().processInstanceId(processInstanceId).singleResult();
+
+    //then
+    assertThat(job.getRetries()).isEqualTo(3);
+
+    // when
+    try {
+      managementService.executeJob(job.getId());
+    } catch (Exception e) {
+      // ignore
+    }
+    job = managementService.createJobQuery().processInstanceId(processInstanceId).singleResult();
+
+    // then
+    assertThat(job.getRetries()).isEqualTo(4);
+  }
+
+
+  @Test
+  void testRetryIntervalsWithFailure() {
+    // given
+    String retryInterval = "PT10M,PT17M,PT20M";
+    String processDefinitionName = "testRetryIntervalsWithFailure";
+    BpmnModelInstance bpmnModelInstance = getBpmnModelInstance(processDefinitionName, retryInterval);
+    testRule.deploy(bpmnModelInstance);
+
+    // when
+    String processInstanceId = runtimeService.startProcessInstanceByKey(processDefinitionName).getId();
+    Job job = managementService.createJobQuery().processInstanceId(processInstanceId).singleResult();
+
+    // then
+    assertThat(job.getRetries()).isEqualTo(3);
+
+    // when
+    try {
+      managementService.executeJob(job.getId());
```

Comment:
use `assertThatThrownBy()` here
`job.getId()` must be declared in the given part as variable